src/query/api/v1/handler/database/kvstore.go

```diff
@@ -162,10 +163,15 @@ func (h *KeyValueStoreHandler) update(
 		return nil, err
 	}
 
-	if err := jsonpb.UnmarshalString(string([]byte(update.Value)), newProto); err != nil {
+	if err := jsonpb.UnmarshalString(string(update.Value), newProto); err != nil {
 		return nil, err
 	}
 
+	oldProtoMarshalled := bytes.NewBuffer(nil)
+	if err := (&jsonpb.Marshaler{}).Marshal(oldProtoMarshalled, oldProto); err != nil {
+		return nil, fmt.Errorf("failed to marshal old proto: %w", err)
+	}
```

Comment:
Hm actually should this error out if it couldn't get previous, or just log a warning? Otherwise potential you get stuck on a poisoned message and can't roll forward, e.g. if we change the format of the proto at some point?

Reply:
hmm good call, will do

Reply:
ah yeah there's already a similar message above